sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TimeTravelSpec.scala

```diff
@@ -38,21 +38,25 @@ object TimeTravelSpec {
       val ts = timestamp.get
       assert(ts.resolved && ts.references.isEmpty && !SubqueryExpression.hasSubquery(ts))
       if (!Cast.canAnsiCast(ts.dataType, TimestampType)) {
-        throw QueryCompilationErrors.invalidTimestampExprForTimeTravel(ts)
+        throw QueryCompilationErrors.invalidTimestampExprForTimeTravel(
```

Comment:
Merge with the bottom logic to form a `INVALID_TIME_TRAVEL_TIMESTAMP_EXPR.DATA_TYPE` error message.